cpp/src/arrow/builder.h

```diff
@@ -720,6 +720,16 @@ class ARROW_EXPORT StringBuilder : public BinaryBuilder {
 
   Status Append(const std::vector<std::string>& values,
                 const uint8_t* valid_bytes = NULLPTR);
+
+  /// \brief Append a sequence of C strings in one shot
+  /// \param[in] values a contiguous C array of C strings
+  /// \param[in] length the number of values to append
+  /// \param[in] valid_bytes an optional sequence of bytes where non-zero
+  /// indicates a valid (non-null) value
+  /// \return Status
+  Status Append(const char** values,
```

Comment:
I'd rather have it called `AppendCStrings` as @xhochy proposed. Too many overloads can create ambiguity.

Reply:
Umm. We already use `Append()` for scalar append and vector append in other builders.
If we use separated names for append variations, it may be better that we also change the current vector append methods to `AppendValues()` or something for consistency.

@xhochy what do you think about this?

Reply:
I would agree to rename other vector append method to `AppendValues`, to avoid ambiguities and user errors, but I don't know how much we value API stability at this point. @xhochy ?

(for example, in Python `list.append` always appends a single item and `list.extend` must be called to append multiple items at once)

Reply:
I'm happy to have methods with better names. We should deprecate the old variants then and leave them in for one release. Removing methods completely between release without deprecation already creates a lot of overhead in the libraries consuming Arrow. While there are not that many public visible, I would like to keep maintenance for them as smooth as possible.

Reply:
Ok, I think we can do this in a separate PR.